src/test/scala/org/ergoplatform/appkit/ergotool/ErgoToolSpec.scala

```diff
@@ -221,6 +261,38 @@ class ErgoToolSpec extends PropSpec with Matchers with ScalaCheckDrivenPropertyC
     res should include ("\"transactionId\": \"ded098c633a7bc145ba87dfa58ae9fde8be252d17aa36fbe734c7cb3f57bbaf3\",")
   }
 
+  property("dex:SellOrder - not enough tokens") {
+    val data = MockData(
+      Seq(
+        loadNodeResponse("response_Box1.json"),
+        loadNodeResponse("response_Box2.json"),
+        loadNodeResponse("response_Box3.json"),
+        loadNodeResponse("response_Box4.json"),
+        "21f84cf457802e66fb5930fb5d45fbe955933dc16a72089bf8980797f24e2fa1"),
+      Seq(
+        loadExplorerResponse("response_boxesByAddressUnspent.json")))
+    val res = runCommandWithCtxStubber("dex:SellOrder",
+      args = Seq(
+        "storage/E2.json",
+        "50000000", // token price in NanoERGs
+        "21f84cf457802e66fb5930fb5d45fbe955933dc16a72089bf8980797f24e2fa1", // tokenId
+        "60", // token amount
+        "5000000" // DEX fee
+      ),
+      expectedConsoleScenario =
+        s"""Storage password> ::abc;
+           |""".stripMargin, 
+      data, 
+      { ctx: BlockchainContext =>
+          val spiedCtx = spy(ctx)
+          val boxes = new java.util.ArrayList[InputBox](0)
+          when(spiedCtx.getUnspentBoxesFor(any[Address])) thenReturn boxes
```

Comment:
@aslesarenko My mocking exploration eventually led me to this attempt to partially mock BlockchainContext inside FileMockedErgoClient. It does not work yet (NPE on this line). I'll continue to dig into it. However, I'm not sure this is a best way to unit test a command.

Reply:
@aslesarenko I fixed partial mocking.